xla/literal_comparison_test.cc

```diff
@@ -25,173 +25,115 @@ limitations under the License.
 namespace xla {
 namespace {
 
-TEST(LiteralComparisonTest, F8E4M3FNCompareNear_Equal) {
-  auto actual =
-      LiteralUtil::CreateR0<tsl::float8_e4m3fn>(tsl::float8_e4m3fn(8.0));
-  auto expected =
-      LiteralUtil::CreateR0<tsl::float8_e4m3fn>(tsl::float8_e4m3fn(8.0));
+template <typename T>
+struct TypeTraits {
+  static constexpr float expected_1ulp = 9.0;
+  static constexpr int within_n_values_1ulp = 1;
+  static constexpr float expected_4ulp = 12.0;
+  static constexpr float expected_using_float_4ulp = 12.1;
+};
+
+template <>
+struct TypeTraits<tsl::float8_e4m3fn> : public TypeTraits<void> {
+  static constexpr PrimitiveType primitive_type = PrimitiveType::F8E4M3FN;
```

Comment:
Instead of having this field, you can just have `GetPrimitiveType` below call [`NativeToPrimitiveType`](https://github.com/openxla/xla/blob/536ba0b7d74f6637a7a772471a99ecf4f578aef2/xla/primitive_util.h#L78)

Reply:
fixed